src/hotspot/share/opto/castnode.cpp

```diff
@@ -506,6 +506,12 @@ const Type* ConstraintCastNode::widen_type(const PhaseGVN* phase, const Type* re
   if (!phase->C->post_loop_opts_phase()) {
     return res;
   }
+
+  // Keep these casts for verification
+  if (VerifyConstraintCasts) {
```

Comment:
Can we have 2 modes, one where we do the widening, and one where we do not?

Reply:
Done